packages/config/src/bin/serve-chrome.ts

```diff
@@ -0,0 +1,164 @@
+const REPO_NAME = 'insights-chrome';
+const CONTAINER_PORT = 8000;
+export const EXPOSED_PORT = 9999;
+
+type ContainerRuntime = 'docker' | 'podman';
+let execBin: ContainerRuntime | undefined = undefined;
+
+const chromeDeploymentConfig = {
+  repo: 'git@gitlab.cee.redhat.com:service/app-interface.git',
+  deployFile: 'data/services/insights/frontend-base/deploy.yml',
+  stablePath: 'resourceTemplates[0].targets[4].ref',
+  tarTarget: path.resolve(__dirname, 'chrome-deploy/'),
+};
+
+const checkoutCommand = `git archive --remote=${chromeDeploymentConfig.repo}  HEAD ${chromeDeploymentConfig.deployFile} |  tar xvf - -C ${chromeDeploymentConfig.tarTarget}`;
+
+function checkContainerRuntime(): ContainerRuntime {
+  try {
+    if (execSync('which podman').toString().trim().length > 0) {
+      return 'podman';
+    }
+    if (execSync('which docker').toString().trim().length > 0) {
+      return 'docker';
+    }
+  } catch (error) {
+    throw new Error('No container runtime found');
+  }
+
+  throw new Error('No container runtime found');
+}
+
+async function getLatestCommits(): Promise<string> {
+  const { data } = await axios.get(`https://api.github.com/repos/${REPO_OWNER}/${REPO_NAME}/commits`, {
+    headers: {
+      Accept: 'application/vnd.github+json',
+      'X-GitHub-Api-Version': '2022-11-28',
+    },
+    params: {
+      per_page: 1,
+    },
+  });
+  if (data.length === 0) {
+    throw new Error('No commits for chrome found!');
+  }
+
+  const { sha } = data[0];
+  return sha.substring(0, 7);
+}
+
+type Deployment = {
+  resourceTemplates: {
+    name: string;
+    targets: {
+      ref: string;
+      namespace: {
+        $ref: string;
+      };
+    }[];
+  }[];
+};
+
+async function getProdRelease() {
+  try {
+    if (!fs.existsSync(chromeDeploymentConfig.tarTarget)) {
+      fs.mkdirSync(chromeDeploymentConfig.tarTarget);
+    }
+    execSync(checkoutCommand, {
+      stdio: 'inherit',
+    });
+
+    const deployment = load(
+      fs.readFileSync(path.join(chromeDeploymentConfig.tarTarget, chromeDeploymentConfig.deployFile), {
+        encoding: 'utf-8',
+      })
+    ) as Deployment;
+    const chromeProd = deployment.resourceTemplates
+      .find((template) => {
+        return template.name === 'insights-chrome';
+      })
+      ?.targets.find((target) => {
+        return target.namespace.$ref.includes('prod-frontends');
+      });
+    if (!chromeProd) {
+      throw new Error('Unable to find chrome prod deployment configuration.');
+    }
+
+    return chromeProd.ref.substring(0, 7);
+  } catch (error) {
+    fecLogger(LogType.error, error);
+    fecLogger(LogType.warn, 'Unable to find chrome prod deployment! Falling back to latest image.');
+    return getLatestCommits();
+  }
+
+  // const { data: deploymentYaml } = await axios.get(deploymentSource);
+  // console.log(deploymentYaml);
+}
+
+function pullImage(tag: string) {
+  execSync(`${execBin} pull quay.io/cloudservices/insights-chrome-frontend:${tag}`, {
```

Comment:
Hardcoded repo.

Reply:
Yes and the server port as well. These need to configurable. I am working on that.